lib/routes/objects.js

```diff
@@ -220,41 +268,98 @@ module.exports =
         /**
          * @type {FastifyHandler}
          */
+        // eslint-disable-next-line max-statements
         async (req, res) => {
           const {
             body: {
               name,
               version,
               data,
-              env = 'production',
+              env: envOrAlias,
               variant = '_default',
               expiration = null
             }
           } = req;
 
-          await fastify.putObjectVariant({
-            name,
-            version,
-            env,
-            data,
-            variant,
-            expiration
-          });
-
-          if (fastify.log.security) {
-            fastify.log.security({
-              success: true,
-              message: `Object "${name}" variant "${variant}" sucessfully created in "${env}"`,
-              // eslint-disable-next-line max-len
-              category: 'database', // see https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html#ecs-event-category-database
-              // eslint-disable-next-line max-len
-              type: ['creation', 'allowed'], // see https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html#ecs-event-type-creation
-              method: req.method,
-              url: req.url,
-              requestId: req.id,
-              sourceAddress: req.ip,
-              host: req.hostname
+          if (envOrAlias) {
+            let env = envOrAlias;
+            let forceCreateEnv = true;
+            const envAlias = await fastify.getEnvAlias({
+              name,
+              alias: envOrAlias
             });
+
+            if (envAlias) {
+              env = envAlias.env;
+              forceCreateEnv = false;
+            }
+
+            await fastify.putObjectVariant(
+              {
+                name,
+                version,
+                env,
+                data,
+                variant,
+                expiration
+              },
+              { forceCreateEnv }
+            );
+
+            if (fastify.log.security) {
+              fastify.log.security({
+                success: true,
+                message: `Object "${name}" variant "${variant}" sucessfully created in "${env}"`,
+                // eslint-disable-next-line max-len
+                category: 'database', // see https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html#ecs-event-category-database
+                // eslint-disable-next-line max-len
+                type: ['creation', 'allowed'], // see https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html#ecs-event-type-creation
+                method: req.method,
+                url: req.url,
+                requestId: req.id,
+                sourceAddress: req.ip,
+                host: req.hostname
+              });
+            }
```

Comment:
```suggestion
              fastify.log.security?.({
                success: true,
                message: `Object "${name}" variant "${variant}" sucessfully created in "${env}"`,
                // eslint-disable-next-line max-len
                category: 'database', // see https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html#ecs-event-category-database
                // eslint-disable-next-line max-len
                type: ['creation', 'allowed'], // see https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html#ecs-event-type-creation
                method: req.method,
                url: req.url,
                requestId: req.id,
                sourceAddress: req.ip,
                host: req.hostname
              });
```

Reply:
I will change all the occurrences like you suggested with a sperate PR. Out of scope in this one.